test/python/primitives/containers/test_bit_array.py

```diff
@@ -695,3 +695,214 @@ def test_expectation_values(self):
+                    [4, 38, 111, 24, 82, 181, 132],
+                    [0, 76, 4, 224, 65, 204, 128],
+                    [5, 104, 97, 194, 23, 80, 32],
+                    [5, 8, 68, 2, 21, 16, 160],
+                    [0, 40, 45, 62, 3, 8, 180],
+                    [4, 65, 0, 20, 16, 97, 22],
+                    [4, 42, 43, 222, 83, 4, 52],
+                    [1, 8, 0, 212, 69, 4, 20],
+                    [0, 73, 36, 50, 67, 109, 166],
+                    [4, 40, 41, 220, 19, 0, 20],
+                    [0, 110, 103, 196, 67, 245, 144],
+                    [0, 99, 71, 4, 64, 117, 146],
+                    [1, 79, 34, 34, 7, 200, 34],
+                    [1, 72, 32, 2, 71, 101, 32],
+                    [1, 106, 111, 42, 7, 121, 160],
+                    [1, 70, 102, 20, 6, 208, 148],
+                    [0, 32, 33, 228, 66, 45, 16],
+                    [5, 68, 4, 208, 84, 229, 132],
+                    [0, 102, 71, 194, 64, 245, 160],
+                    [4, 0, 76, 234, 80, 28, 160],
+                    [0, 111, 43, 232, 67, 237, 2],
+                    [1, 41, 105, 234, 7, 57, 34],
+                    [0, 46, 99, 50, 3, 185, 36],
+                    [1, 110, 79, 58, 5, 249, 164],
+                    [1, 104, 97, 2, 7, 113, 32],
+                    [5, 69, 108, 62, 86, 253, 182],
+                    [0, 44, 97, 194, 3, 144, 32],
+                    [1, 69, 100, 212, 70, 212, 150],
+                    [5, 79, 14, 238, 85, 204, 178],
+                    [1, 8, 100, 6, 71, 20, 176],
+                    [1, 74, 98, 198, 71, 117, 48],
+                    [4, 77, 68, 242, 17, 216, 166],
+                    [0, 76, 76, 58, 65, 253, 164],
+                    [4, 35, 107, 248, 82, 61, 6],
+                    [1, 98, 43, 46, 70, 109, 48],
+                    [0, 98, 107, 44, 2, 88, 16],
+                    [0, 7, 6, 54, 0, 169, 182],
+                    [4, 8, 100, 52, 19, 24, 148],
+                    [1, 106, 99, 34, 71, 125, 32],
+                    [5, 32, 101, 22, 86, 20, 180],
+                    [5, 5, 40, 10, 86, 132, 34],
+                    [0, 10, 2, 192, 65, 37, 0],
+                    [5, 12, 76, 56, 85, 189, 132],
+                    [5, 1, 32, 214, 22, 0, 54],
+                ],
+                dtype="uint8",
+            ),
+            num_bits=num_bits,
+        )
+
+        bitpairs = np.array(
+            [
+                [16, 43],
+                [27, 35],
+                [17, 37],
+                [10, 22],
+                [31, 30],
+                [12, 38],
+                [20, 50],
+                [33, 41],
+                [14, 46],
+                [29, 11],
+                [8, 13],
+                [26, 4],
+                [2, 28],
+                [34, 7],
+                [45, 32],
+                [40, 1],
+                [18, 48],
+                [42, 15],
+                [25, 5],
+            ],
+            dtype=int,
+        )
+
+        np.random.seed(0)
```

Comment:
Thank you for implementing this new feature.
Could you use `Generator` (`np.random.default_rng`) instead of `RandomState` (`np.random`)?

> [Generator](https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.Generator) and its associated infrastructure was introduced in NumPy version 1.17.0. There is still a lot of code that uses the older [RandomState](https://numpy.org/doc/stable/reference/random/legacy.html#numpy.random.RandomState) and the functions in [numpy.random](https://numpy.org/doc/stable/reference/random/index.html#module-numpy.random). While there are no plans to remove them at this time, we do recommend transitioning to [Generator](https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.Generator) as you can.

https://numpy.org/doc/stable/reference/random/index.html#module-numpy.random

```suggestion
        rng = np.random.default_rng(0)
```

Reply:
Ended up replacing the randomized test with smaller ones without randomness.